src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/design_patterns/DesignPatternFactory.java

```diff
@@ -0,0 +1,35 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.design_patterns;
+
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DesignPatternFactory {
+    public static class DesignPatternNotRegisteredException extends Exception {
+        DesignPatternNotRegisteredException(String s) {
+            super(s);
+        }
+    }
+
+    private static Map<String, DesignPattern> registeredDesignPatterns = new HashMap<>();
```

Comment:
[Naming]
keyToValue or valueByKey pattern is more describing for maps.
designPatternByName or nameToDesignPattern.